examples/moving_least_squares/symmetric_pseudoinverse_svd.hpp

```diff
@@ -0,0 +1,229 @@
+/****************************************************************************
+ * Copyright (c) 2023 by the ArborX authors                                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#pragma once
+
+#include <Kokkos_Core.hpp>
+
+#include <cassert>
+#include <cmath>
+#include <limits>
+
+// Pseudo-inverse moment matrix using SVD
+// We must find U, E (diagonal and positive) and V such that A = U.E.V^T
+// We also know that A is symmetric (by construction), so U = SV where S is
+// a sign matrix (only 1 or -1 in the diagonal, 0 elsewhere).
+// Thus A = U.E.S.U^T
+template <class ValueType, typename MemorySpace>
+class SymmPseudoInverseSVD
```

Comment:
Maybe we should move this in a separate PR, and then add some tests for it?

Reply:
I'm also a bit confused. Why is this a class, and why do we need to store data? It's called exactly once from MLS, and I don't think we use any of the stored `_es`, `_inv`, etc, anywhere in MLS. Why not make it a free function?

Reply:
> Maybe we should move this in a separate PR, and then add some tests for it?
Why not, as it is one of the more complex set of functions and could actually be used for other things.

> I'm also a bit confused. Why is this a class, and why do we need to store data? It's called exactly once from MLS, and I don't think we use any of the stored _es, _inv, etc, anywhere in MLS. Why not make it a free function?
This is usually how I create complex algorithms that need temporary variables. The user cannot instantiate it so these variables are dropped as soon as the static method returns. However, I can transform it so it uses only free functions.

Reply:
I have modified this part so it resembles what one could find in `ArborX::Details`. The inverse is now computed using only free functions and with less restrictions on the input.